mirrord/layer/src/detour.rs

```diff
@@ -147,7 +144,7 @@ pub(crate) enum Bypass {
     /// We hooked a file operation on a path in mirrord's bin directory. So do the operation
     /// locally, but on the original path, not the one in mirrord's dir.
     #[cfg(target_os = "macos")]
-    FileOperationInMirrordBinTempDir(*const c_char),
+    FileOperationInMirrordBinTempDir(PathBuf),
```

Comment:
Why do you want to return a new `PathBuf` here?
Why not the new pointer like before, or just the length of the removed prefix?

Reply:
We can't return a pointer into the original `String` because the switch over to using `PathBuf` means that operations like stripping a prefix result in a new `PathBuf`, where before they kept the same `String` buffer - so the `FileOperationInMirrordBinTempDir` needs to own the value, otherwise the pointer will be dangling

Reply:
I disagree. You don't need to return a `PathBuf`, as all the information the callers (that handle that variant) ever need, is how many bytes of the path to skip (this is equivalent to a new pointer into the original path buffer, either one works). If we still want to create new `PathBuf`s (which I don't think is necessary) while finding out that number, we can still at the end just check the length difference of the old and new `PathBuf`s, and only return that number.